dango/dex/src/core/liquidity_pool.rs

```diff
@@ -254,6 +306,98 @@ impl PassiveLiquidityPool for PairParams {
 
         Ok((reserve, input))
     }
+
+    fn reflect_curve(
+        &self,
+        base_denom: Denom,
+        quote_denom: Denom,
+        reserves: &CoinPair,
+    ) -> StdResult<(
+        impl Iterator<Item = StdResult<(Udec128, Uint128)>>,
+        impl Iterator<Item = StdResult<(Udec128, Uint128)>>,
+    )> {
```

Comment:
@pacmanifold I'm thinking instead of returning `StdResult<(Udec128, Uint128)>` we just return `(Udec128, Uint128)`.

Firstly, this makes your code simpler: no need for the `unwrap_or_return_err_as_option!` macro.

Secondly, if there's an error in reflecting the curve, it only means the iterator returns `None`, meaning no more order is placed, instead of throwing an error in `cron_execute` which reverts all the changes. Generally, we should ensure `cron_execute` never fails.

Thoughts?